src/main/java/jenkins/bouncycastle/api/BcProviderRegistration.java

```diff
@@ -0,0 +1,46 @@
+package jenkins.bouncycastle.api;
+
+import java.security.Security;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Utility class for registration of the BouncyCastle security provider.
+ */
+@Restricted(NoExternalUse.class)
```

Comment:
Since the class isn't public this restriction shouldn't be necessary, but it doesn't hurt to be extra explicit about it.